compiler/src/iree/compiler/Codegen/LLVMGPU/ConvertToROCDL.cpp

```diff
@@ -90,6 +104,16 @@ struct ConvertToROCDLPass final
   void runOnOperation() override {
     ModuleOp m = getOperation();
 
+    m.walk(
+        [&](Operation *op) {
+          if (failed(validateDataTypes(op))) {
+            op->emitOpError()
+                << "F8E5M2 and F8E4M3FN types are not supported on "
+                   "the ROCm backend; try F8E5M2FNUZ or F8E4M3FNUZ instead.";
```

Comment:
The exact error message can be emitted from within the validation function. This is so that the error and the validation logic stay together and don't diverge